src/if-run/builtins/time-sync/index.ts

```diff
@@ -131,20 +137,31 @@ export const TimeSync = (
             .diff(compareableTime)
             .as('seconds');
 
-          /** Checks if there is gap in timeline. */
+          validateIntervalForResample(
+            input.duration,
+            timeParams.upsamplingResolution
+          );
+
           if (timelineGapSize > 1) {
+            /** Checks if there is gap in timeline. */
             acc.push(
               ...getZeroishInputPerSecondBetweenRange(
                 compareableTime,
                 currentMoment,
-                input
+                input,
+                timeParams.upsamplingResolution
```

Comment:
Hi @mouhamadalmounayar, thank you for your contribution! 
While checking I've noticed that `getZeroishInputPerSecondBetweenRange` function already got 4 function arguments. Let's change the function signature and make it to accept two arguments: `input` and `params`. In this case `params` will include `compareableTime`, `currentMoment` and `upsamplingResolution` properties.